documents/compute_fields.go

```diff
@@ -2,11 +2,14 @@ package documents
 
 import (
 	"bytes"
+	"context"
+	"time"
 
+	coredocumentpb "github.com/centrifuge/centrifuge-protobufs/gen/go/coredocument"
 	"github.com/centrifuge/go-centrifuge/errors"
 	"github.com/centrifuge/go-substrate-rpc-client/scale"
 	logging "github.com/ipfs/go-log"
-	"github.com/perlin-network/life/exec"
+	"github.com/wasmerio/go-ext-wasm/wasmer"
```

Comment:
had to change to wasmer since perlin doesn't have execution control to stop the function execution